### DISCUSSION TITLE
Display subtitles below picture

### DISCUSSION THREAD

**Discussion by w-sky** [smplayer-dev/smplayer#647] [Ideas]
Most movies are wider than a 16:9 or 16:10 screen thus when watching a movie, there is usually black space below and above the picture. And I think there, below the picture is the perfect place to display subtitles.

I can move the subtitles up and down, but at "100" which is at the bottom within the picture, it stops and can't go further down.

**Comment by smplayer-dev** [comment]
Menu Video -> Filters -> Add black borders.

**Comment by Galathorne** [comment]
So I have been using SMPlayer since around 2007. Every few months I get irritated again that I cannot get the subtitles and OSD to display on the black borders, and then I scour the internet form answers, going down multiple rabbit holes. Every time. I have not been able to fix this and it still shows up directly on the video, no matter how big the black borders are on the top and bottom.

To be clear, I have:

1. tried to add black borders from the filters menu
2. checked "Add black borders..." from the Setting > Options > General > Video page (both of them)
3. played around with every conceivable setting (Direct rendering, Video slices, etc, etc.)
4. meticulously checked smplayer.ini and mpv.conf for any issues, so much so that they are permanently loaded when I open Notepad++
6. read, reread and again read the online mpv manual for any issues that might interfere with SMPlayer
7. started from scratch with a new SMPlayer portable and manually changed every setting again
8. Asked chatGPT, Claude.ai, Bard, Phind and some other AI chat bots, all giving different but equally non-working solutions

After all these years, I have not been able to get my subtitles and OSD to get off of the (insert swear word here) video frame and onto the black borders that is just sitting there empty. I'm starting to think that personal curses are a thing.

For more context, I have used Linux (various), Windows 7, 8 and now Windows 10. Never made a difference.
I have been using SMPlayer portable, updating whenever an update is available. I prefer the portable version so havent checked an installed setup in ages, so if it is working when installed, then this is a bug in the portable version. But then - I struggle to believe that with all the updates over all the years there would be the same inconsistency between an installation and portable version, so I don't think it has something to do with that.

Can anybody help me shed light on what is going on and how I can fix this issue? I can share any and all config file contents if that will help.

**Reply by Galathorne** [reply]
Not the place for this. I use SMPlayer BECAUSE it is available in both Windows and Linux. However, no matter my reason or your reason for sharing your promotion, this thread is about fixing an issue in SMPlayer, not asking for alternatives for a single OS.

**Comment by smplayer-dev** [comment]
Try disabling hardware decoding in preferences -> performance. That option disables all video filters.

**Comment by Galathorne** [comment]
Thanks so much. That is the first time I can get the OSD off the video.

However:

1. The subtitles are still where they were, not moved to the black bars at the bottom.
2. The video is moved down and cropped at the bottom.
3. The OSD is not at the top of the screen. I have set it to be 5px from the top, and it is doing something strange by only starting a ways down.

Here is a capture of my full screen, with both OSD and subtitles:

![image](https://github.com/smplayer-dev/smplayer/assets/87099108/13a132de-e77a-48b5-8309-b5a9bacabeeb)

And here is the same frame capture in mpv with the full video height. Notice the bottom black border is the same height as on SMPlayer:

![image](https://github.com/smplayer-dev/smplayer/assets/87099108/e6356c38-e023-438c-a6d2-95d75d00047c)

So, SMPlayer moved the video down to accommodate the OSD at the top but cropped it at the bottom black bar, so to speak, without doing anything with the subtitles.

I do run mpv scripts, but disabling them completely has no effect on the outcome as described here.

At this stage, I am not sure which settings in my smplayer.ini and mpv.conf can be interfering with one another, as I am so saturated by reading and trying all the options. If I paste them both here, could someone take a look—perhaps try them on your side and see if you can figure it out?

**Comment by smplayer-dev** [comment]
What's the subtitle format of that video?
Subtitles in SRT format should move to the bottom but I'm not sure about SSA/ASS.

**Comment by bt4ibwem8** [comment]
@smplayer-dev 
I am using only SRT subtitles format but there are many movies which do not move subs into black box. I have to add that "black borders" filter if i wanna see the subtitles under the movie. However if i try this movie to play in official mpv gui, then it works fine in the black box.
So my assumption is it is some bug in Smplayer...

**Comment by smplayer-dev** [comment]
Try enabling the option "add black borders on fullscreen" in preferences -> general -> video.

**Comment by bt4ibwem8** [comment]
It works. Do you know why the subtitles (in default settings) are sometimes shown in the bottom black box and sometimes do not? You think it depens on which movie is playing?

**Reply by smplayer-dev** [reply]
Maybe some movies already have black bars in the picture.

**Comment by reisa211** [comment]
Hello guys! I really want to use SMplayer but I'm facing a problem that stops me using it! The idea is to up and down the whole picture and place the subtitles at the very bottom. This picture is from celluloid!

![Screenshot from 2024-05-29 10-53-41](https://github.com/smplayer-dev/smplayer/assets/163755261/60797eab-80fa-443d-96af-014d5901531c)






I've managed to up and down the picture in smplayer with the option in the config file "allow_video_movement=true"! I've managed to place the subs in the black bar with "add black borders on fullscreen" in preferences -> general -> video"! But unlike the celluloid I can't place the subs over 100 and when you enlarge the subs they show in the picture again!

![Screenshot from 2024-05-29 10-57-40](https://github.com/smplayer-dev/smplayer/assets/163755261/db35b351-21c9-4fda-9881-1546151f258b)

How can I fix this?

**Reply by Galathorne** [reply]
I've tried every possible setting and configuration change, but the issue persists. If you've decided to overlook the problem for the sake of familiarity, it might be best to get used to this behavior. However, if you're not willing to tolerate it, it may be time to switch to a different player that meets your needs. Personally, I doubt this issue with SMPlayer will be resolved anytime soon (not since 2007), but I'll hold out a faint glimmer of hope and continue using it.

**Reply by reisa211** [reply]
Yap, I've already switched to celluloid! I've made 2 config files and now I'm super happy! The first config is for subs and picture UP and DOWN.
F1 add sub-pos 1
F2 add sub-pos -1
F3 add sub-scale -0.01
F4 add sub-scale 0.01
F5 add video-pan-y 0.010
F6 add video-pan-y -0.010

And the second config file is for downmixing 5.1 and 7.1 to srereo!
## Downmix 5.1/7.1 audio ##

[Downmix_Audio_5_1]

profile-cond=get("audio-params/channel-count") >= 5 and get("audio-params/channel-count") < 7

profile-restore=copy-equal

volume-max=200

af=lavfi="lowpass=c=LFE:f=120,volume=1.6,pan=stereo|FL=0.5*FC+0.707*FL+0.707*BL+0.5*LFE|FR=0.5*FC+0.707*FR+0.707*BR+0.5*LFE"

[Downmix_Audio_7_1]

profile-cond=get("audio-params/channel-count") >= 7

profile-restore=copy-equal

volume-max=200

af=lavfi="lowpass=c=LFE:f=120,volume=1.6,pan=stereo|FL=0.5*FC+0.3*FLC+0.3*FL+0.3*BL+0.3*SL+0.5*LFE|FR=0.5*FC+0.3*FRC+0.3*FR+0.3*BR+0.3*SR+0.5*LFE"

Now, the sound and subs are perfect! I've managed to resolve the problem for 1 day! 17 years bro... hahahahahaha GG!

**Reply by smplayer-dev** [reply]
> Hello guys! I really want to use SMplayer but I'm facing a problem that stops me using it! The idea is to up and down the whole picture and place the subtitles at the very bottom. This picture is from celluloid!

Add something like this in preferences -> advanced -> mpv -> video filters:
`lavfi=[pad=1920:1080:(1920-ow)/2:(1080-oh)/2]`
